src/m365/spo/commands/site/site-recyclebinitem-list.ts

```diff
@@ -0,0 +1,101 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+  type?: string;
+  secondary?: boolean;
+}
+
+class SpoSiteRecycleBinItemListCommand extends SpoCommand {
+  public get name(): string {
+    return commands.SITE_RECYCLEBINITEM_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists items from recycle bin';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['Id', 'Title', 'DirName'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    if (this.verbose) {
+      logger.logToStderr(`Retrieving all items from recycle bin at ${args.options.siteUrl}...`);
+    }
+
+    const state: string = args.options.secondary ? '2' : '1';
+
+    let requestUrl: string = `${args.options.siteUrl}/_api/site/RecycleBin?$filter=(ItemState eq ${state})`;
+
+    if (typeof args.options.type !== 'undefined') {
+      const type = SpoSiteRecycleBinItemListCommand.recycleBinItemType.find(item => item.value === args.options.type);
+      if (typeof type !== 'undefined') {
+        requestUrl += ` and (ItemType eq ${type.id})`;
+      }
+    }
+
+    const requestOptions: any = {
+      url: requestUrl,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    request
+      .get<any>(requestOptions)
+      .then((value: any[] ): void => {
+        logger.log(value);
```

Comment:
Let's remove the `value` wrapper from the response to keep the output consistent with other commands. So instead of `value`, we should log `value.value` (or better rename the first `value` to `response` to make the code easier to follow).

Reply:
done